src/manifolds/ProbabilitySimplex.jl

```diff
@@ -281,6 +281,49 @@ mean(::ProbabilitySimplex, ::Any...)
 
 default_estimation_method(::ProbabilitySimplex, ::typeof(mean)) = GeodesicInterpolation()
 
+@doc raw"""
+    rand(::ProbabilitySimplex; vector_at=nothing, σ::Real=1.0)
+
+
+When `vector_at` is `nothing`, return a random (uniform) point `x` on the [`ProbabilitySimplex`](@ref)
+manifold `M` by a stick-breaking processes, i.e. sample an ordered set of breakpoints on the unit
+interval and return the sequential differences.
+
+When `vector_at` is not `nothing`, return a (Gaussian) random vector from the tangent space
+``T_{vector\_at}\mathrm{\Delta}^n``by shifting a multivariate Gaussian with standard deviation `σ` 
+to have a zero component sum.
+"""
+rand(::ProbabilitySimplex; σ::Real=1.0)
+
+function Random.rand!(M::ProbabilitySimplex, pX; vector_at=nothing, σ=one(eltype(pX)))
+    n = only(representation_size(M))
+    if isnothing(vector_at)
+        _r = sort(vcat(rand(n - 1), [0.0, 1.0]))
+        pX .= (circshift(_r, -1) .- _r)[1:n]
+    else
+        _r = randn(n) .* σ
+        pX .= _r .- mean(_r)
```

Comment:
We don't need the allocation to `r_`.
```suggestion
       Random.randn!(pX)
       pX .= (pX .- mean(pX)) .* σ
```